script.js

```diff
@@ -1,4 +1,435 @@
+// set global variables
+// to track games and progress
+var numGamesDraw = 0;
+var numGamesWon = 0;
+var numGamesPlayed = 0;
+
+// to set game mode
+var currentGameMode = "enter user name";
+console.log(currentGameMode);
+
+// to set user name
+var userName = "";
+
+// for Korean game
+var lastWinner = 0;
+
+//set function to return percentage win
+var percentageWin = function () {
+  var winRate = Math.floor((numGamesWon / numGamesPlayed) * 100);
+  return winRate;
+};
+
+// function to generate random computer move
+var randomItem = function () {
+  var random = Math.floor(Math.random() * 3);
+  var programMove = "stone";
+  if (random == 1) {
+    programMove = "scissors";
+  }
+  if (random == 2) {
+    programMove = "paper";
+  }
+  return programMove;
+};
+
+// function to determine when game is a draw where computer's move and players move is the same.
+var drawGame = function (input, computerMove) {
+  return input == computerMove;
+};
+
+// function to determine when player wins in normal mode
+var playerWin = function (input, computerMove) {
+  return (
+    (input == "scissors" && computerMove == "paper") ||
+    (input == "stone" && computerMove == "scissors") ||
+    (input == "paper" && computerMove == "stone")
+  );
+};
+
+// function to determine when player wins in reversed mode
+// created another var so that we do not need to change the number of games and won percentage logic etc
+var playerWinReserved = function (input, computerMove) {
+  return (
+    (input == "stone" && computerMove == "paper") ||
+    (input == "paper" && computerMove == "scissors") ||
+    (input == "scissors" && computerMove == "stone")
```

Comment:
There is something know as [SCREAM_CASE constants ](https://www.freecodecamp.org/news/when-to-capitalize-your-javascript-constants-4fabc0a4a4c4/)

I can touch more on this as the days go by, but it's simply another way to name variables that remain unchanged through out the program eg.
```
var STONE = "stone"
var SCISSORS = "scissors"
var PAPER = "paper"
```